front/lib/plans/workspace_usage.ts

```diff
@@ -22,3 +24,42 @@ export async function countActiveSeatsInWorkspace(
     },
   });
 }
+
+/**
+ * An active user is a user who has posted at least one message during the billing period.
+ * To handle Slack users, we are using the userContextEmail field of the message to distinguish users.
+ */
+export async function countActiveUsersInWorkspaceSince({
+  workspace,
+  since,
+}: {
+  workspace: WorkspaceType;
+  since: Date;
+}): Promise<number> {
+  const result = await front_sequelize.query<{
+    count: string;
+  }>(
+    `
+      SELECT COUNT(DISTINCT "userMessage"."userContextEmail") as "count"
```

Comment:
How costly is that query? because even if we're not awaiting on it we're going to run it a lot.

The alternative is to start a temporal workflow that we debounce with a large debouncing window?

Reply:
@spolu we could run it only once a day, we don't need to have the usage up to date at all times on Stripe, we actually need it just before the end of the billing period. 

The temporal workflow is much more work.

Reply:
How would you run it once a day only?

Reply:
First message of the day?

Reply:
if I understand correctly, this is using redis to debounce

https://github.com/dust-tt/dust/pull/2451/files#diff-9f8cedbec664679e963523519adc65f135e241c37d44edd85b1b9987c54d6026R485 


however Im not sure I fully understand how we avoid losing information from when we skip updating because the key is already in redis ?

Reply:
We could ask the Redis key to live a bit longer than 1 hour, but today we make the call max every 1/hour if there's a new message, seems okay?

Indeed, with this quick solution we might loose some users if they are posting their first message right after someone else and then no one talks until the end of the billing period. Seems okay but I can do the temporal workflow if you both think this is not good!